File: neetcode/wip/06_linked_list/MergeTwoSortedLinkedLists.java
package neetcode;

public class MergeTwoSortedLinkedLists {
    /**
     * Definition for singly-linked list.
     */
    public class ListNode {
        int val;
        ListNode next;

        ListNode() {
        }

        ListNode(int val) {
            this.val = val;
        }

        ListNode(int val, ListNode next) {
            this.val = val;
            this.next = next;
        }
    }

    public static ListNode mergeTwoLists(ListNode list1, ListNode list2) {
        return new ListNode();
    }

    public static void main(String[] args) {
        mergeTwoLists();
//        neetcode.DepthOfBinaryTree.TreeNode a = new neetcode.DepthOfBinaryTree.TreeNode(10);
//        neetcode.DepthOfBinaryTree.TreeNode b = new neetcode.DepthOfBinaryTree.TreeNode(5, new neetcode.DepthOfBinaryTree.TreeNode(7), new neetcode.DepthOfBinaryTree.TreeNode(8));
//        neetcode.DepthOfBinaryTree.TreeNode root = new neetcode.DepthOfBinaryTree.TreeNode(1, a, b);
//        System.out.println("Should be 3: " + neetcode.DepthOfBinaryTree.maxDepth(root));
    }
}

File: neetcode/wip/06_linked_list/merge-two-sorted-linked-lists.py
from typing import Optional


class ListNode:
    """Definition for singly-linked list."""

    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next


class Solution:
    def mergeTwoLists(self, list1: ListNode, list2: ListNode) -> ListNode:

        output = ListNode()

        while True:
            if list1.val < list2.val:
                output.next = ListNode(list1.val)

            if list1.next is None and list2.next is None:
                break

        return output


# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
# Test cases
# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

sol = Solution()

cases = [
    # My test cases
    # (),
    # NeetCode test cases
    ([1, 1, 2, 3, 4, 5], ([1, 2, 4], [1, 3, 5])),
    ([1, 2], ([], [1, 2])),
    ([], ([], [])),
]

# Run tests
for expected, _input in cases:
    print(_input)
    list1 = ListNode(_input[0][0])
    for i in range(1, len(_input[0])):
        list1.next = ListNode(_input[0][i])
    actual = sol.mergeTwoLists(_input[0], _input[1])
    print("assert {expected} == {actual}")
    print(f"assert {expected} == {actual}")
    print()
    assert expected == actual
